clients/pkg/promtail/targets/file/filetargetmanager.go

```diff
@@ -273,6 +272,25 @@ func (s *targetSyncer) sync(groups []*targetgroup.Group, targetEventHandler chan
 	targets := map[string]struct{}{}
 	dropped := []target.Target{}
 
+	// process target removal first to avoid cleaning up fileEventWatchers that are needed by a new or modified target
+	for key, target := range s.targets {
+		if _, ok := targets[key]; !ok {
```

Comment:
this will never evaluate to OK since targets above is empty, I need to rework this before it is ready for review